tour_model_eval/user_input_request_times_all_users.ipynb

```diff
@@ -0,0 +1,568 @@
+    "\n",
+    "# get valid user list\n",
+    "user_ls,valid_users = evaluation.get_user_ls(all_users,radius)\n",
+    "\n",
+    "for a in range(len(all_users)):\n",
+    "    user = all_users[a]\n",
+    "    filter_trips,sim,trips = evaluation.filter_data(user,radius)\n",
+    "    logging.debug(\"len(filter_trips)is %s \"% len(filter_trips))\n",
+    "\n",
+    "    # filter out users that don't have enough valid labeled trips\n",
+    "    if not evaluation.valid_user(filter_trips,trips):\n",
+    "        req_day.append(NaN)\n",
+    "        req_month.append(NaN) \n",
+    "        continue\n",
+    "    sim.bin_data()\n",
+    "    sim.delete_bins()\n",
+    "    bins = sim.bins\n",
+    "    \n",
+    "    # collect requested trips and common trips(no need to request) indices above cutoff\n",
+    "    ab_trip_ls = []\n",
+    "    no_req_trip_ls = []\n",
+    "    for bin in bins:\n",
+    "        early_trip = filter_trips[bin[0]]\n",
+    "        index = 0\n",
+    "        for i in range(1,len(bin)):\n",
+    "            compare_trip = filter_trips[bin[i]]\n",
+    "            if early_trip['data']['start_local_dt']['year']>compare_trip['data']['start_local_dt']['year']:\n",
+    "                early_trip = compare_trip\n",
+    "                index = i\n",
+    "            elif early_trip['data']['start_local_dt']['year']==compare_trip['data']['start_local_dt']['year'] and early_trip['data']['start_local_dt']['month']>compare_trip['data']['start_local_dt']['month']:\n",
+    "                early_trip = compare_trip\n",
+    "                index = i\n",
+    "            elif early_trip['data']['start_local_dt']['year']==compare_trip['data']['start_local_dt']['year'] and early_trip['data']['start_local_dt']['month']==compare_trip['data']['start_local_dt']['month'] and early_trip['data']['start_local_dt']['day']>compare_trip['data']['start_local_dt']['day']:\n",
+    "                early_trip = compare_trip\n",
+    "                index = i\n",
+    "        early_trip_index = bin[index]\n",
+    "        ab_trip_ls.append(early_trip_index)\n",
+    "        \n",
+    "        for k in range(len(bin)):\n",
+    "            if k != index:\n",
+    "                no_req_trip_idx = bin[k]\n",
+    "                no_req_trip_ls.append(no_req_trip_idx)\n",
+    "\n",
+    "\n",
+    "    \n",
+    "    # bins below cutoff\n",
+    "    bl_bins = sim.below_cutoff\n",
+    "    \n",
+    "    # collect requested trips indices below cutoff\n",
+    "    bl_trip_ls = []\n",
+    "    for bin in bl_bins:\n",
+    "        for trip_index in bin:\n",
+    "            bl_trip_ls.append(trip_index)\n",
+    "            \n",
+    "    # whole list of requested trips indices\n",
+    "    req_trips_ls=ab_trip_ls+bl_trip_ls\n",
+    "    \n",
+    "    \n",
+    "    # collect request times in a day\n",
+    "    bin_day = evaluation.bin_date(req_trips_ls,filter_trips,day=True)\n",
```

Comment:
I'm confused; I don't see this method in `confirmed_trips_eval_bins_clusters.py`